src/openrct2/ride/Ride.cpp

```diff
@@ -2092,26 +2092,26 @@ TrackColour ride_get_track_colour(Ride* ride, int32_t colourScheme)
     return result;
 }
 
-vehicle_colour ride_get_vehicle_colour(Ride* ride, int32_t vehicleIndex)
+VehicleColour ride_get_vehicle_colour(Ride* ride, int32_t vehicleIndex)
 {
-    vehicle_colour result;
+    VehicleColour result;
 
     // Prevent indexing array out of bounds
     vehicleIndex = std::min<int32_t>(vehicleIndex, MAX_CARS_PER_TRAIN);
 
-    result.main = ride->vehicle_colours[vehicleIndex].Body;
-    result.additional_1 = ride->vehicle_colours[vehicleIndex].Trim;
-    result.additional_2 = ride->vehicle_colours[vehicleIndex].Ternary;
+    result.Body = ride->vehicle_colours[vehicleIndex].Body;
```

Comment:
You actually don't need to create this temporary you can just return the rides vehicle colour directly